projects/scion/toolkit/operators/src/operators.ts

```diff
@@ -10,17 +10,25 @@
 
 import {map, mergeMap, share, switchMap, take} from 'rxjs/operators';
 import {combineLatest, concat, EMPTY, from, identity, MonoTypeOperatorFunction, noop, Observable, Observer, of, OperatorFunction, pipe, SchedulerLike, Subscriber, Subscription, TeardownLogic} from 'rxjs';
-import {Arrays} from '@scion/toolkit/util';
+import {Arrays, Observables} from '@scion/toolkit/util';
 
 /**
  * Filters items in the source array and emits an array with items satisfying given predicate.
  *
  * If passing `undefined` as predicate, the filter matches all items.
  */
 export function filterArray<T, S extends T>(predicate?: (item: T) => item is S): OperatorFunction<T[], S[]>;
-export function filterArray<T>(predicate?: (item: T) => boolean): MonoTypeOperatorFunction<T[]>;
-export function filterArray<T>(predicate?: (item: T) => boolean): MonoTypeOperatorFunction<T[]> {
-  return map((items: T[]): T[] => items.filter(item => !predicate || predicate(item)));
+export function filterArray<T>(predicate?: (item: T) => Observable<boolean> | Promise<boolean> | boolean): MonoTypeOperatorFunction<T[]>;
+export function filterArray<T>(predicate?: (item: T) => Observable<boolean> | Promise<boolean> | boolean): MonoTypeOperatorFunction<T[]> {
+  if (!predicate) {
+    return identity;
+  }
+  const NO_MATCH = {};
+  return pipe(
+    mapArray(item => Observables.coerce(predicate(item)).pipe(map(matches => matches ? item : NO_MATCH as any))),
```

Comment:
A predicate returning a Promise or Observable is a very particular use case. If all predicates return a boolean, which is the default case, perhaps we can omit `combineLatest`?